cmd/data-scanner.go

```diff
@@ -1240,7 +1240,7 @@ func applyExpiryOnNonTransitionedObjects(ctx context.Context, objLayer ObjectLay
 		}
 	}()
 
-	dobj, err = objLayer.DeleteObject(ctx, obj.Bucket, obj.Name, opts)
+	dobj, err = objLayer.DeleteObject(ctx, obj.Bucket, encodeDirObject(obj.Name), opts)
```

Comment:
I can see this is encoded inside erasureServerPools.DeleteObject() as well, I do not think this is needed

Reply:
@vadmeste it is not - it is doing extra work see why this is needed from Decommission code.